src/udev/udev-builtin-blkid.c

```diff
@@ -231,6 +232,15 @@ static int find_gpt_root(UdevEvent *event, blkid_probe pr, const char *loop_back
                 /* We found a root partition, nice! Let's export its UUID. */
                 if (!sd_id128_is_null(root_id))
                         udev_builtin_add_property(event, "ID_PART_GPT_AUTO_ROOT_UUID", SD_ID128_TO_UUID_STRING(root_id));
+
+                /* Also encode current factory reset state in udev property at the moment we are probing
+                 * this. This is relevant because in various contexts we'd rather not create gpt-auto
+                 * symlinks while a factory reset is ongoing. */
+                FactoryResetMode f = factory_reset_mode();
```

Comment:
Rather than introduce a udev property, how about to support `CONST{factory_reset}="..."` token in udev-rules.c ?
I do not think it is better to save the mode in udev database.
And only provided for block devices. It is completely system-wide status, rather than per-device status.

Reply:
CONST{} sounds wrong, because the factory reset state is *not* constant after all. it might be on initially, but eventually once everything is completed is turned off. a udev property can model this really nicely: it first is there and pins the situation in a readable way on the device (i.e. for each device you can see whether we were in fr state or not when we probed.)

i mean, we coul split this into a separate builtin if you like?

Reply:
Indeed, it is not constant. But, system wide. Yeah, introducing another built-in sounds better to me.

But, as said in the above, I am not sure if it is really necessary that we should create two kinds of symlinks.